features/networking/pod.feature

```diff
@@ -190,3 +190,153 @@ Feature: Pod related networking scenarios
+    And evaluation of `@result[:response].strip` is stored in the :master_ip clipboard
+    
+    Given I select a random node's host
+    Given I have a project
+    And SCC "privileged" is added to the "system:serviceaccounts:<%= project.name %>" group
+    When I run the :create client command with:
+      | f | https://raw.githubusercontent.com/weliang1/Openshift_Networking/master/egress-http-proxy/egress-http-proxy-pod.yaml |
+    Then the pod named "egress-http-proxy" becomes ready
+
+    #Pod should not access MCS via an egress router pod
+    When I execute on the pod:
+      | curl | -I | https://<%= cb.master_ip %>:22623/config/master | -k |
+    Then the output should contain "Connection refused"
+    When I execute on the pod:
+      | curl | -I | https://<%= cb.master_ip %>:22624/config/master | -k |
+    Then the output should contain "Connection refused"
+
+  # @auther anusaxen@redhat.com
+  # @case_id OCP-23893
+  @admin
+  Scenario: A pod in a namespace with an egress IP cannot access the MCS
+    Given I use the first master host
+    #Step to obtain master IP
+    And I run commands on the host:
+      | hostname -i |
+    Then the step should succeed
+    And evaluation of `@result[:response].strip` is stored in the :master_ip clipboard
+    
+    Given I select a random node's host
+    And evaluation of `node.name` is stored in the :egress_node clipboard
+    #add the egress ip to the hostsubnet
+    And the valid egress IP is added to the "<%= cb.egress_node %>" node
+    Given I have a project
+    And evaluation of `project.name` is stored in the clipboard
+    # add the egress ip to the project
+    When I run the :patch admin command with:
+    | resource      | netnamespace                         |
+    | resource_name | <%= project.name %>                    |
+    | p             | {"egressIPs":["<%= cb.valid_ip %>"]} |
+    | type          | merge                                |
+    Then the step should succeed
+    #pod-for-ping will be a non-hostnetwork pod
+    And I have a pod-for-ping in the project
+    
+    #Pod cannot access MCS
+    When I execute on the pod:
+      | curl | -I | https://<%= cb.master_ip %>:22623/config/master | -k |
+    Then the output should contain "Connection refused"
+    When I execute on the pod:
+      | curl | -I | https://<%= cb.master_ip %>:22624/config/master | -k |
+    Then the output should contain "Connection refused"
+
+  # @auther anusaxen@redhat.com
+  # @case_id OCP-23894
+  @admin
+  Scenario: User cannot access the MCS by creating a service that maps to non-MCS port to port 22623 or 22624 on the IP of a master (via manually-created ep's)
+    Given I use the first master host
+    #Step to obtain master IP
+    And I run commands on the host:
+      | hostname -i |
+    Then the step should succeed
+    And evaluation of `@result[:response].strip` is stored in the :master_ip clipboard
+    
+    Given I select a random node's host
```

Comment:
actually where do you use this host?

Reply:
I don't understand. If you mean the master's host, its IP is being used in patch step.

Reply:
This step sets variable `@host` as well sets current `node`. I don't see you use either of those later in scenario so I'm asking whether we can remove this step?

Reply:
Thanks i understand now. Yes, this step can be removed as my intention was to execute commands and create pods on a compute node instead of master and without this step too, that will be achieved.

Reply:
Would you remove then? @zhaozhanqi , could you tell if your concerns have been addressed? Can we merge? PR seems to be mostly complete.

Reply:
Done! @akostadinov we may have to hold merge for couple of biz days as developer Dan Winship suggested to test egress case in a different way so i might need to submit tweaks next week hence WIP is added to this PR. I will remove the WIP as soon as things are cleared with developer and changed submitted to this PR post that.

Reply:
And its getting delayed due to vpshere and bare metal shared env is down today. So possibly need to be looked on Monday

Reply:
@akostadinov @anuragthehatter 
Most of them are good for me except the egress-router case, please help update it according the comment of cases review. thanks.